kiwi_lint/generic_foreign_key.py

```diff
@@ -0,0 +1,34 @@
+import astroid
+from pylint.checkers import utils, BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint_django.utils import node_is_subclass
+
+
+class GenericForeignKeyChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+
+    name = 'generic-foreign-key-checker'
+
+    msgs = {
+        'R4525': (
+            'Avoid using GenericForeignKey',
+            'avoid-generic-foreign-key',
+            'Avoid using django.contrib.contenttypes.fields.GenericForeignKey',
+        )
+    }
+
+    @utils.check_messages('avoid-generic-foreign-key')
+    def visit_classdef(self, node):
+
+        if not node_is_subclass(node, 'django.db.models.base.Model'):
+            return
+
+        for part in node.body:
+            if not isinstance(part, astroid.Assign):
+                continue
+
+            if not isinstance(part.value, astroid.Call):
+                continue
```

Comment:
when do we need to check Call nodes ? Can you give an example ? 

I think checking assignments only (e.g. name = <FieldClass>) where FieldClass inherits from GenericForeignKey would be enough.

Reply:
I am not sure what I was tested with when I got an error.
I have removed the check.